File: sample/UseMaven/GNAdSamples/javaSample/src/main/java/jp/co/geniee/samples/BaseMenuActivity.java
package jp.co.geniee.samples;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;
import android.widget.TextView;

public abstract class BaseMenuActivity extends AppCompatActivity {

    private ListView mListView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initValues();

        setupListViewContents(getListViewContents());
    }

    private void initValues() {
        mListView = findViewById(R.id.listView);
    }

    private void setupListViewContents(final MenuItem[] items) {
        ArrayAdapter<MenuItem> listAdapter = new ArrayAdapter<MenuItem>(this, R.layout.simple_list_item, items) {
            @Override
            public View getView(int position, View convertView, ViewGroup parent) {
                View row = convertView;
                if (row == null) {
                    LayoutInflater inflater = (LayoutInflater) this.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
                    row = inflater.inflate(R.layout.simple_list_item, parent, false);
                }

                MenuItem item = items[position];

                TextView title = row.findViewById(R.id.text1);
                title.setText(item.getTitle());
                TextView subtitle = row.findViewById(R.id.text2);
                subtitle.setText(item.getSubtitle());

                return row;
            }
        };

        mListView.setAdapter(listAdapter);

        AdapterView.OnItemClickListener itemClickListener = new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                Intent intent = items[position].getIntent();

                if (intent != null) startActivity(intent);
            }
        };
        mListView.setOnItemClickListener(itemClickListener);
    }

    protected abstract MenuItem[] getListViewContents();
}

File: sample/kotlinSample/src/main/java/jp/co/geniee/samples/BaseMenuActivity.kt
package jp.co.geniee.samples

import android.content.Context
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ListView
import android.widget.TextView

abstract class BaseMenuActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)

        setupListViewContents(getListViewContents())
    }

    private fun setupListViewContents(items: Array<MenuItem>) {
        val listAdapter = object : ArrayAdapter<MenuItem>(this, R.layout.simple_list_item, items) {
            override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
                var row = convertView
                if (row == null) {
                    val inflater = this.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
                    row = inflater.inflate(R.layout.simple_list_item, parent, false)
                }

                val item = items[position]

                val title = row!!.findViewById<TextView>(R.id.text1)
                title.text = item.title
                val subtitle = row.findViewById<TextView>(R.id.text2)
                subtitle.text = item.subTitle

                return row
            }
        }

        val listView = findViewById<ListView>(R.id.listView)
        listView.adapter = listAdapter

        val itemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val intent = items[position].intent

            startActivity(intent)
        }
        listView.onItemClickListener = itemClickListener
    }

    abstract fun getListViewContents(): Array<MenuItem>

}
